test/parallel/test-assert-fail.js

```diff
@@ -67,5 +67,5 @@ common.expectsError(() => {
 // The stackFrameFunction should exclude the foo frame
 assert.throws(
   function foo() { assert.fail('first', 'second', 'message', '!==', foo); },
-  (err) => !/foo/m.test(err.stack)
+  (err) => !/foo(?!\/)/m.test(err.stack)
```

Comment:
BTW, it seems that `/m` flag is redundant here, as it is useful only if `^$` symbols are used.